Wrappers/Python/ccpi/web_viewer/trame_viewer.py

```diff
@@ -420,84 +252,42 @@ def construct_slice_window_range_slider(self):
+            disabled=disabled,
             label="Slice window",
             thumb_label=True,
             style=style,
         )
 
-    def construct_color_slider(self):
-        if self.cmax > 100:
-            # Use actual values
-            min_value = self.cmin
-            max_value = self.cmax
-            step = 1
-            self.color_slider_is_percentage = False
-        else:
-            # Use percentages
-            min_value = 0
-            max_value = 100
-            step = 0.5
-            self.color_slider_is_percentage = True
-
-        return vuetify.VRangeSlider(label="Color range",
-                                    hide_details=True,
-                                    solo=True,
-                                    v_model=("coloring", self.windowing_defaults),
-                                    min=min_value,
-                                    max=max_value,
-                                    step=step,
-                                    disabled=self.disable_3d,
-                                    thumb_label=True,
-                                    style="max-width: 300px")
-
-    def construct_windowing_slider(self):
-        if self.cmax > 100:
-            # Use actual values
-            min_value = self.cmin
-            max_value = self.cmax
-            step = 1
-            self.windowing_slider_is_percentage = False
+    def update_slice_data(self):
+        if hasattr(self.cil_viewer, "getSliceMapWindow"):
+            self.cmin, self.cmax = self.cil_viewer.getSliceMapWindow((0., 100.))
+            self.slice_window_range_defaults = self.cil_viewer.getSliceMapWindow((60., 95.))
         else:
-            # Use percentages
-            min_value = 0
-            max_value = 100
-            step = 0.5
-            self.windowing_slider_is_percentage = True
-
-        return vuetify.VRangeSlider(label="Windowing",
-                                    hide_details=True,
-                                    solo=True,
-                                    v_model=("windowing", self.windowing_defaults),
-                                    min=min_value,
-                                    max=max_value,
-                                    step=step,
-                                    disabled=self.disable_3d,
-                                    thumb_label=True,
-                                    style="max-width: 300px")
-
-    def update_windowing_defaults(self, method="scalar"):
-        self.cmin, self.cmax = self.cil_viewer.getVolumeMapWindow((0., 100.), method)
-        self.windowing_defaults = self.cil_viewer.getVolumeMapWindow((80., 99.), method)
-        self.slice_window_range_defaults = self.cil_viewer.getVolumeMapWindow((5., 95.), "scalar")
+            self.cmin, self.cmax = self.cil_viewer.getVolumeMapWindow((0., 100.), "scalar")
+            self.slice_window_range_defaults = self.cil_viewer.getVolumeMapWindow((5., 95.), "scalar")
```

Comment:
The slice window range defaults should not come from the VolumeMapWindow

Reply:
We should replace this whole if/else with:

```python
ia = vtk.vtkImageHistogramStatistics()
ia.SetInputData(self.cil_viewer.voi.GetOutput())
ia.SetAutoRangePercentiles(0, 100)
ia.Update()
self.cmin, self.cmax = ia.GetAutoRange()
ia.SetAutoRangePercentiles(5.0, 95.)
ia.Update()
self.slice_window_range_defaults = ia.GetAutoRange()
ia.Update()
```

What this does is it sets cmin and cmax to the min and max values for the SLICE. In the 2D viewer that is all you need. In the 3D viewer, these are different to the min and max value for the entire image volume.

To get the min and max value for the image volume you need:
`cmin, cmax = self.cil_viewer.getVolumeMapWindow((0., 100.), "scalar") `, like you have at line 266.

So what you will need for the 3D viewer is to store two different sets of cmin and cmax, one which is for the slice slider and one which is for the volume slider. And the slice slider range needs updating whenever the slice number is changed.

Reply:
What I will do later is add the code for getting the cmin and cmax for the slice and getting the window range defaults for the slice as methods in both the 2D and 3D viewer (with the same names), so that later we can call these instead of you having to use all this code in the web app.

Reply:
But these additions to the viewer will be done after this PR is merged, just to clarify.